docs/DesignDocs/SGX2-EDMM.md

```diff
@@ -0,0 +1,105 @@
+# SGX2 EDMM in Open Enclave SDK
+
+Intel SGX2 on Icelake CPUs supports enclave dynamic memory management (EDMM),
+which many applications of the Open Enclave SDK (OE) will want to take advantage
+of. This document outlines the strategy for making these features available to
+applications.
+
+The key mechanism to support the dynamic addition of memory pages is the
+`ENCLS[EAUG]` instruction, which allocates and zeroes a new memory page via
+communication with the SGX driver. Before the enclave uses the new page, it
+acknowledges its addition by issuing an `ENCLU[EACCEPT]` instruction for the
+respective page address. Further, the `ENCLS[EMODPR]` and `ENCLU[EMODPE]`
+instructions restrict or extend page permissions, respectively. Page type
+changes (including page removal/trimming) are performed by the `ENCLS[EMODT]`
+instruction.
+
+## Heap allocation
+
+We propose to add a new OCALL to OE enclaves which requests new memory pages
+from the SGX driver (via `ioctl`), combined with an enclave function that, upon
+return from the OCALL, accepts the new pages. This does not require handling of
+signals or page faults and is easily integrated with memory managers like the
+default dlmalloc, by pre-mapping a configurable maximum number of pages and
+tracking the number of currently mapped pages. For performance reasons, we
+propose to add a switchless version of the OCALL whenever
+`OE_ENCLAVE_SETTING_CONTEXT_SWITCHLESS` is enabled:
+
+    trusted {
```

Comment:
Line 25 says OCALL but line 28 says trusted which means ECALL

Reply:
Thanks, that was a typo, it should say `untrusted` of course.